subiquity/server/apt.py

```diff
@@ -257,11 +259,13 @@ async def configure_for_install(self, context):
 
         codename = lsb_release(dry_run=self.app.opts.dry_run)["codename"]
 
-        write_file(
-            self.install_tree.p("etc/apt/sources.list"),
-            f"deb [check-date=no] file:///cdrom {codename} main restricted\n",
-            mode=0o644,
-        )
+        if not new_style_iso:
+            # ISOs for 26.04 and later have the cdrom source enabled in the squashfs.
+            write_file(
+                self.install_tree.p("etc/apt/sources.list"),
+                f"deb [check-date=no] file:///cdrom {codename} main restricted\n",
```

Comment:
not that I particularly want to optimize for this case, but what does it mean for someone to edit (create) the pool to add additional packages?

what if, instead of detecting the "new style iso" by a sentinel filename in `sources.list.d`, we detect instead by looking for `/cdrom/dists/pool`, and we add this entry if that directory is present.    Adding packages to a new style iso then becomes the somewhat simpler exercise of adding a package pool to the expected location, which I think anyone who's modifying an existing pool should be able to handle.

In that case, we kind of don't care if it's a new style ISO or not, we just add the pool if it's present, which should handle old and new cases.